helix-core/src/snippets/elaborate.rs

```diff
@@ -0,0 +1,370 @@
+use std::mem::swap;
+use std::ops::Index;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
```

Comment:
What do you think about avoiding anyhow in `helix-core`? To me anyhow seems useful for stuff like helix-term and commands but for library-like stuff like helix-core it makes it easy to not introduce custom Error types. Since we only have one usage here it would be easy to drop and replace with something thin like a struct + thiserror

Reply:
Sounds good